### PR TITLE
Json encoder customization

### LINKED ISSUE #196: Not possible to return Decimals from the controller
I'm getting a serialization error when returning python Decimal data types with the built in automatic JSON serialization.
I guess one solution would be to add special handling in JSONEncoder and convert to str or float or maybe just make it possible to provide a custom JSON serializer.

  File "/usr/local/lib/python3.4/dist-packages/connexion/decorators/produces.py", line 44, in default
    return json.JSONEncoder.default(self, o)
  File "/usr/lib/python3.4/json/encoder.py", line 173, in default
    raise TypeError(repr(o) + " is not JSON serializable")
TypeError: Decimal('1') is not JSON serializable

### PR DESCRIPTION
**Pull request by rafaelcaricio**
Fixes #196.

Connexion should remain as must compatible as possible with usual Flask applications. This PR adds back the support of developers to customize the [json_encoder](http://flask.pocoo.org/snippets/119/) in Connexion applications exactly as in pure Flask apps.

Changes proposed in this pull request:

 - Make use the flask [`JSONEncoder`](http://flask.pocoo.org/docs/0.10/api/#flask.json.JSONEncoder) internally in Connexion.
 - Relay on [`flask.Flask.json_encoder`](http://flask.pocoo.org/docs/0.10/api/#flask.Flask.json_encoder) for customization of encoding. This makes possible for users of Connexion to override the encoder they wanna use.
 - Adds documentation for this feature.

### CODE REVIEW

**Comment by jmcs** [comment]
:+1:

File: connexion/decorators/produces.py [hunk 1]
```diff
@@ -124,7 +124,7 @@ def wrapper(*args, **kwargs):
                 logger.debug('Endpoint returned an empty response (204)', extra={'url': url, 'mimetype': self.mimetype})
                 return '', 204, headers
 
-            data = json.dumps(data, indent=2, cls=JSONEncoder)
+            data = json.dumps(data, indent=2)
```
Comment (hjacobs) [inline connexion/decorators/produces.py]: How is the custom JSON encoder now used by json.dumps?

Comment (jmcs) [inline connexion/decorators/produces.py]: I had the same question, but somehow it's working on the tests.

Comment (rafaelcaricio) [inline connexion/decorators/produces.py]: This `json.dumps` come from `flask` namespace (http://flask.pocoo.org/docs/0.10/api/#flask.json.dumps) which changes the behaviour to use as encoder the class referenced by the flask app attribute `json_encoder`. That is the expected behaviour while using JSON encoders with flask. So you can personalize the encoder changing the `app.json_encoder`.

References:
http://flask.pocoo.org/snippets/119/
http://flask.pocoo.org/docs/0.10/api/#flask.Flask.json_encoder

**Comment by rafaelcaricio** [comment]
Can we merge that?

**Comment by jmcs** [comment]
@rafaelcaricio if you rebase the branch, yes.

**Comment by rafaelcaricio** [comment]
@jmcs Didn't realize it got conflicts. Should be good now. :+1:

**Comment by rafaelcaricio** [comment]
@jmcs Didn't realize it got conflicts. Should be good now. :+1:

**Pull request merged**